pkg/ipam/crd.go

```diff
@@ -538,3 +538,11 @@ func (a *crdAllocator) Dump() (map[string]string, string) {
 	status := fmt.Sprintf("%d/%d allocated", len(allocs), a.totalPoolSize())
 	return allocs, status
 }
+
+// TriggerRefresh updates the custom resource in the apiserver based on the latest
+// information in the local node store
+func (a *crdAllocator) TriggerRefresh(reason string) error {
```

Comment:
Let's call this `SyncUpstream()` to be more generic. The concept of refreshing the node is clear for the CRD IPAM where there is a CiliumNode CRD but it is not clear from a general-purpose IPAM perspective.